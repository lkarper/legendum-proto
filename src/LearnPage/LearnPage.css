.LearnPage__container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    height: calc(100vh - 125px);
}

.LearnPage__image {
    position: absolute;
    bottom: 25px;
    margin: 0 auto;
    left: 0;
    right: 0;
    max-height: 75%
}

.LearnPage__background-container {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    margin: 0;
    height: 50%;
}

.LearnPage__background-image {
    z-index: -10;    
    height: 100%;
    width: 100%;
    display: block;
}

.LearnPage__text-hints-container {
    width: 100%;
    display: flex;
    flex-direction: column;
}

.LearnPage__text-container {
    position: absolute;
    top: 50%;
    left: 0;
    min-height: 50%;
    width: 100vw;
    padding: 5px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background: url('/images/paper-background-2-small.png') repeat-y center center fixed;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
}

.LearnPage__do-link {
    align-self: center;
}

.LearnPage__text {
    margin: 0;
    padding: 5px;
}

.LearnPage__button-container {
    /* position: absolute; */
    width: 100%;
    /* bottom: 10px; */
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

.LearnPage__nav-button {
    width: 75px;
}