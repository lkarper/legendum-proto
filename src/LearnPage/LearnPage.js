import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import LearnHints from '../LearnHints/LearnHints';
import BackgroundImage from '../BackgroundImage/BackgroundImage';
import './LearnPage.css';

const LearnPage = (props) => {
    const [showHintsBoolean, setShowHintsBoolean] = useState(false);

    const { pages, page, setPage, chapt } = props.data;
    const pageToDisplay = pages[page - 1];

    const image = pageToDisplay.image_url
        ?
            <BackgroundImage
                classPrefix='LearnPage'
                imgUrl={pageToDisplay.background_image_url}
                imageAltText={pageToDisplay.background_image_alt_text}
            >
                <img 
                    className='LearnPage__image'
                    src={pageToDisplay.image_url} 
                    alt={pageToDisplay.image_alt_text} 
                />
            </BackgroundImage>
        :
            <BackgroundImage 
                classPrefix='LearnPage'
                imgUrl={pageToDisplay.background_image_url}
                imageAltText={pageToDisplay.background_image_alt_text}
            />;

    return (
        <div className='LearnPage__container'>
            {image}
            <div className='LearnPage__text-container'>
                <div className='LearnPage__text-hints-container'>
                    <p className='LearnPage__text'>{pageToDisplay.text}</p>
                    {pageToDisplay.hints 
                        ? 
                            <LearnHints 
                                page={pageToDisplay} 
                                showHintsBoolean={showHintsBoolean} 
                                setShowHintsBoolean={setShowHintsBoolean} 
                            />
                        : ''
                    }
                    {page === pages.length 
                        ?
                            <Link
                                className='LearnPage__do-link' 
                                to={`/game/exercises/${chapt}/do`}
                            >
                                Practice What You've Learned
                            </Link>
                        : 
                            ''
                        }
                </div>
                <div className='LearnPage__button-container'>
                    <button
                        className='LearnPage__nav-button'
                        disabled={page === 1} 
                        onClick={() => {
                            setShowHintsBoolean(false);
                            setPage(page - 1);
                        }}
                    >
                        &#60;
                    </button>
                    <button
                        className='LearnPage__nav-button'
                        disabled={page === pages.length} 
                        onClick={() => {
                            setShowHintsBoolean(false);
                            setPage(page + 1);
                        }}
                    >
                        &#62;
                    </button>    
                </div>
            </div>
        </div>
    );
};

export default LearnPage;