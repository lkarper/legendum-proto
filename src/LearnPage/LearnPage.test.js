import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LearnPage from './LearnPage';

describe('LearnPage component', () => {
  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <LearnPage />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const wrapper = mount(
      <BrowserRouter>
        <LearnPage />
      </BrowserRouter>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
