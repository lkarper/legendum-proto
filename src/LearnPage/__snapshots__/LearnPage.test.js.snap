// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnPage component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LearnPage
      chapt=""
      page={1}
      pages={Array []}
      setPage={[Function]}
    >
      <p>
        Error: Looks like something went wrong. Check your connection and try
      </p>
    </LearnPage>
  </Router>
</BrowserRouter>
`;
